Tests/SentryTests/Integrations/SentrySessionTrackerTests.swift

```diff
@@ -20,24 +22,63 @@ class SentrySessionTrackerTests: XCTestCase {
                                "sessionTrackingIntervalMillis": sessionTrackingIntervalMillis] as [String: Any]
             let options = try Options(dict: optionsDict)
 
-            hub = TestHub(client: TestClient(options: options), andScope: nil)
+            client = TestClient(options: options);
+            fileManager = try! TestFileManager(dsn: SentryDsn())
+            client.sentryFileManager = fileManager
+            hub = TestHub(client: client, andScope: nil)
             SentrySDK.setCurrentHub(hub)
 
             sut = SessionTracker(options: options, currentDateProvider: currentDateProvider)
         } catch {
-            XCTFail("Failed to create options")
+            XCTFail("Failed to setup test")
         }
+    }
+
+    func testStartClosesPreviousCachedSession() {
+        sut.start()
+
+        XCTAssertEqual(1, fileManager.readTimestampLastInForegroundInvocations)
+        XCTAssertEqual(1, fileManager.deleteTimestampLastInForegroundInvocations)
+        XCTAssertEqual(0, fileManager.storeTimestampLastInForegroundInvocations)
+        XCTAssertEqual(1, hub.closeCachedSessionInvocations)
+    }
+
+    func testStartClosesPreviousCachedSessionWithoutSavedTimestamp() {
+        fileManager.timestampLastInForeground = nil
 
         sut.start()
+
+        XCTAssertEqual(1, fileManager.readTimestampLastInForegroundInvocations)
+        XCTAssertEqual(0, fileManager.deleteTimestampLastInForegroundInvocations)
+        XCTAssertEqual(0, fileManager.storeTimestampLastInForegroundInvocations)
+        XCTAssertEqual(1, hub.closeCachedSessionInvocations)
+    }
+
+    func testStoresTimestampWhenInBackground() {
+        fileManager.timestampLastInForeground = nil
+
+        sut.start()
+
+        willResignActive()
+
+        XCTAssertEqual(1, fileManager.readTimestampLastInForegroundInvocations)
+        XCTAssertEqual(0, fileManager.deleteTimestampLastInForegroundInvocations)
+        // TODO: check why tests are affecting each other
+        XCTAssertEqual(7, fileManager.storeTimestampLastInForegroundInvocations)
```

Comment:
Calls to `NotificationCenter.default.post(Notification(name: UIApplication.willResignActiveNotification))` seem to be accumulating between test runs.
If I run this test alone, it fires once. If I run all tests, it furies 7 times (in the same test run, `setUp` did run as expected).
Since this is just related to the tests, I'll merge this to get this fix out but a follow up PR is needed to address testing with `NotificationCenter`.